tap_hubspot/client.py

```diff
@@ -47,12 +45,13 @@ def authenticator(self) -> _Auth:
             An authenticator instance.
         """
 
-        access_token = self.config.get("access_token")
-
-        return BearerTokenAuthenticator.create_for_stream(
-            self,
-            token=access_token,
-        )
+        if "refresh_token" in self.config:
+            return HubSpotOAuthAuthenticator.create_for_stream(stream=self)
+        else:
+            return BearerTokenAuthenticator.create_for_stream(
+                self,
+                token=self.config.get("access_token"),
+            )
```

Comment:
```suggestion
            return BearerTokenAuthenticator(
                token=self.config.get("access_token"),
            )
```

Reply:
@edgarrmondragon with this change I get `TypeError: __init__() missing 1 required positional argument: 'stream'`. I think I still need to pass in the `self` as the first arg.